### PR TITLE
backend_template docs and fixes

### LINKED ISSUE #5090: `Template` backend fails when `usetex` is set
The draw-nothing `Template` backend fails when I select `usetex` in the rcparams, with error:

```
AttributeError: 'RendererTemplate' object has no attribute '_text2path'
```

The following minimal example reproduces this for me:

```
import numpy as np

import matplotlib
matplotlib.use('Template')

import matplotlib.pyplot as plt

plt.rc('text', usetex=True)

x = np.linspace(0., 2.*np.pi, 100)
y = np.sin(x)

plt.plot(x,y)
plt.savefig('test.pdf')
```

Looking at the code for the `RendererTemplate` class suggests that there is a missing call to `RendererBase.__init__` in `RendererTemplate.__init__`, but I defer to those with more knowledge of the internals of matplotlib.

### PR DESCRIPTION
**Pull request by anntzer**
Update the docs to be mildly more relevant to third-party backend
implementers.  Add entry for rendered docs.

Make backend_template "work" (in the do-nothing sense) with usetex, by
super-initing the Renderer class and by not assuming the presence of a
`.height` attribute in FigureCanvasBase.

Closes #5090.  xref #12553.

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/backend_bases.py [hunk 1]
```diff
@@ -579,9 +579,10 @@ def _get_text_path_transform(self, x, y, s, prop, angle, ismath):
         path = Path(verts, codes)
         angle = np.deg2rad(angle)
         if self.flipy():
+            width, height = self.get_canvas_width_height()
             transform = Affine2D().scale(fontsize / text2path.FONT_SCALE,
                                          fontsize / text2path.FONT_SCALE)
-            transform = transform.rotate(angle).translate(x, self.height - y)
```
Comment (jklymak) [inline lib/matplotlib/backend_bases.py]: A few tests broke.  Are you sure this change is correct?  Its the only thing I can see that might have broken them...

Comment (anntzer) [reply lib/matplotlib/backend_bases.py]: failures seem spurious now

Comment (timhoffm) [reply lib/matplotlib/backend_bases.py]: Restarted the failing test. The commit looks good overall, but I do not fully understand the change in the height-handling. Thus, at least I want to see that all tests pass.

**Comment by anntzer** [comment]
@dstansby good to merge? :)

**Comment by dstansby** [comment]
Given the force pushes after @timhoffm's review, I think someone else should give it a look over.

**Comment by anntzer** [comment]
I re-requested his review :)

**Pull request merged**